packages/wrangler/src/deploy/index.ts

```diff
@@ -274,8 +281,8 @@ export async function deployHandler(
 	const accountId = args.dryRun ? undefined : await requireAuth(config);
 
 	const assetPaths =
-		args.assets || config.assets
-			? getAssetPaths(config, args.assets)
+		args.legacyAssets || config.legacy_assets
```

Comment:
same comment here: `args.legacyAssets || config.legacy_assets || config.assets`

Reply:
nvm. rawConfig.assets and rawConfig.legacy_assets are merged and normalized into config.legacy_assets.